packages/ui-react/src/components/defi/chain-tvl-historical-chart.tsx

```diff
@@ -0,0 +1,127 @@
+import { useQuery } from "@tanstack/react-query";
+import { useMemo } from "react";
+import { CartesianGrid, Line, LineChart, XAxis, YAxis } from "recharts";
+import {
+	type ChartConfig,
+	ChartContainer,
+	ChartTooltip,
+	ChartTooltipContent,
+} from "#components/shadcn/chart";
+import { Skeleton } from "#components/shadcn/skeleton";
+
+type ChainTvlHistoricalChartProps = {
+	chainId: string;
+	title?: string;
+	color?: string;
+};
+
+export type DefiLlamaTvlDataPoint = {
+	date: number;
+	tvl: number;
+};
+
+const chartConfig = {} satisfies ChartConfig;
+
+async function getChainTvlHistoricalChartData(
+	chainId: string,
+): Promise<DefiLlamaTvlDataPoint[]> {
+	const response = await fetch(
+		`https://api.llama.fi/v2/historicalChainTvl/${chainId}`,
+	);
+
+	if (!response.ok) {
+		throw new Error(`Failed to fetch TVL data: ${response.statusText}`);
+	}
+
+	const data = await response.json();
+	return data as DefiLlamaTvlDataPoint[];
+}
+
+export function ChainTvlHistoricalChart({
+	chainId,
+	title,
+	color,
+}: ChainTvlHistoricalChartProps) {
+	const { data, isLoading } = useQuery({
+		queryKey: ["chain-tvl-historical-chart", chainId],
+		queryFn: () => getChainTvlHistoricalChartData(chainId),
+	});
+
+	const formatDate = (timestamp: number) => {
+		return new Date(timestamp * 1000).toLocaleDateString();
+	};
+
+	const formatTvl = (value: number) => {
```

Comment:
lets use function at utils and not to couple with this component.

packages/domain/src/amount.ts

you can refer to docs at [`Intl.NumberFormat`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat) to achieve wt you need

```
new Intl.NumberFormat("en-GB", {
  notation: "compact",
  compactDisplay: "short",
}).format(987654321);
```